src/main/scala/xitrum/view/Scalate.scala

```diff
@@ -105,6 +105,57 @@ object ScalateEngine {
     }
     builder.toString
   }
+
+  def getPrecompiledTemplateInstance(relUri: String): Template = {
+    val withDots = relUri.replace('/', '.')
+    val xs = withDots.split('.')
+    val extension = xs.last
+    val baseFileName = xs(xs.length - 2)
+    val prefix = xs.take(xs.length - 2).mkString(".")
+    val className = "scalate." + prefix + ".$_scalate_$" + baseFileName + "_" + extension
+    val klass = CLASS_RESOLVER.resolve(className)
+    ConstructorAccess.get(klass).newInstance().asInstanceOf[Template]
+  }
+
+  private def getPrecompiledScalateViewInstance(model: AnyRef, viewName: String, extension: String): Template = {
+    val classSearchList = new ListBuffer[Class[_]]()
+
+    def buildClassList(clazz: Class[_]): Unit = {
+      if (clazz != null && clazz != classOf[Object] && !classSearchList.contains(clazz)) {
+        classSearchList.append(clazz)
+        buildClassList(clazz.getSuperclass)
+        for (i <- clazz.getInterfaces) {
+          buildClassList(i)
+        }
+      }
+    }
+
+    buildClassList(model.getClass)
+
+    classSearchList.toIterator.map { klass =>
+      val withDots = s"${klass.getName}_${viewName}.${extension}"
+      Try(getPrecompiledTemplateInstance(withDots))
+    }.find(_.isSuccess)
+      .map(_.get)
+      .getOrElse(throw new NoSuchViewException(model, viewName))
+  }
+
+  private def getScalateViewInstance(templateDirUri: String, model: AnyRef, viewName: String, extension: String): String = {
+    val path = model.getClass.getName.replace(".", "/")
+    s"${templateDirUri}/${path}.${viewName}.${extension}"
+  }
+
+  def viewTemplate(model: AnyRef, engine: STE, viewName: String, templateType: String) = {
+    if (Config.productionMode) {
+      getPrecompiledScalateViewInstance(model, viewName, templateType)
+    } else {
+      val uri = getScalateViewInstance(developmentTemplateDirUri, model, viewName, templateType)
+      engine.load(uri, Nil)
+    }
+
+  }
+
+  final val developmentTemplateDirUri = "src/main/scalate"
```

Comment:
Please move this up, together with `CLASS_RESOLVER`, `CONTEXT_BINDING_ID` etc.

And also use upper case for it, like `DEVELOPMENT_TEMPLATE_DIR_URI`, or maybe shorter: `DEV_TEMPLATE_DIR_URI`.